dll/win32/framedyn/wbemtime.cpp

```diff
@@ -0,0 +1,670 @@
+/*
+* COPYRIGHT:       See COPYING in the top level directory
+* PROJECT:         ReactOS system libraries
+* FILE:            dll/win32/framedyn/wbemtime.cpp
+* PURPOSE:         WBEMTime class implementation
+* PROGRAMMERS:     Jin Juanshi (2033537949@qq.com)
+*/
+
+#include <wbemtime.h>
+#include <stdio.h>
+#include <time.h>
+#include <strsafe.h>
+
+#define SECONDS_PER_MINUTE 60
+#define SECONDS_PER_HOUR SECONDS_PER_MINUTE * 60
+#define SECONDS_PER_DAY SECONDS_PER_HOUR * 24
+
+#define MICORSECONDS_OFFSET 10
+#define MILLISECONDS_OFFSET MICORSECONDS_OFFSET * 1000
+#define SECONDS_OFFSET MILLISECONDS_OFFSET * 1000
+
+#define DAYS_BETWEEN_TIME_T_AND_FILETIME 134774 //Between 01-01-1601 and 01-01-1970, there are exactly 134774 days.
+
+static __inline void time_t_to_FILETIME(time_t t, LPFILETIME lpFileTime)
+{
+    LONGLONG ll = ((LONGLONG)t + (LONGLONG)DAYS_BETWEEN_TIME_T_AND_FILETIME * SECONDS_PER_DAY) * SECONDS_OFFSET;
+    lpFileTime->dwLowDateTime = (DWORD)ll;
+    lpFileTime->dwHighDateTime = ll >> 32;
```

Comment:
> It is not recommended that you add and subtract values from the FILETIME structure to obtain relative times. Instead, you should copy the low- and high-order parts of the file time to a ULARGE_INTEGER structure, perform 64-bit arithmetic on the QuadPart member, and copy the LowPart and HighPart members into the FILETIME structure.
>
> Do not cast a pointer to a FILETIME structure to either a ULARGE_INTEGER* or __int64* value because it can cause alignment faults on 64-bit Windows.

Reply:
You could use a LARGE_INTEGER, then do your LONGLONG operations on the QuadPart, and then directly assign the low and high parts to the lpFileTime without writing in an explicit way the conversions & casts.